core/src/main/java/com/ibm/wala/ipa/cha/ClassHierarchy.java

```diff
@@ -706,6 +708,97 @@ private void recursiveStringify(Node n, StringBuilder buffer) {
     }
   }
 
+  // Puts ClassHierarchy into a JSON readable variable that maps the class to a list of its
+  // subclasses
+  public Object toJson() {
+    // lines 713-715: initialize variables (subclass and dag) to store important values like hashmap
+    // of each iteration and a list of all maps of class to subclasses
+    HashMap<Object, Object> subclass = new HashMap<>();
+    Iterator<Node> children = root.getChildren();
+    ArrayList<HashMap<Object, Object>> dag = new ArrayList<>();
+    // lines 717-721: Goes through all subclasses of the root class, and adds the subclass hashmap
+    // to the list of all subclass hashmaps
+    while (children.hasNext()) {
+      Node temp = children.next();
+      HashMap<Object, Object> fin = helper_toJson(temp, subclass);
+      dag.add(fin);
+    }
+    // lines 723-725: Puts all hashmaps into one larger hashmap
+    for (int i = 1; i < dag.size(); i++) {
+      dag.get(0).putAll(dag.get(i));
+    }
+    // lines 727-730: Removes unnecesarry parts from name of the class
+    String key = root.getJavaClass().toString();
+    key = key.replace("<Primordial,", "");
+    key = key.replace("<Application,", "");
+    key = key.replace(">", "");
+    // lines 732-742: inserting the root class to its subclasses into the main hashmap while
+    // removing unnecessary substrings from the class name
+    Iterator<Node> root_children = root.getChildren();
+    Set<String> final_root_children = new HashSet<>();
+    while (root_children.hasNext()) {
+      Node val = root_children.next();
+      String child_name = val.getJavaClass().toString();
+      child_name = child_name.replace("<Primordial,", "");
+      child_name = child_name.replace("<Application,", "");
+      child_name = child_name.replace(">", "");
+      final_root_children.add(child_name);
+    }
+    dag.get(0).put(key, final_root_children);
+    Gson gson = new Gson();
+    return gson.toJson(dag.get(0));
+  }
+
+  // helper function to toJson that performs recursion to go through all of the DAG
+  public HashMap<Object, Object> helper_toJson(Node n, HashMap<Object, Object> hash) {
```

Comment:
Since this function always returns the `hash` parameter that gets passed in, it can just return `void` (the caller will always have access to the map it passed in)